test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -116,16 +116,24 @@ def _broadcast_state_dict(self, model, state_dict):
         # TODO (rohan-varma): remove model
         return _broadcast_state_dict(self.rank, state_dict)
 
-    def _compare_models(self, model, model_new, assert_fn, check_fp16=False):
+    def _state_compare(self, model, model_new, assert_fn, state_generator="parameters"):
+        state_base = list(getattr(model, state_generator)())
+        state_new = list(getattr(model_new, state_generator)())
+        # Regardless of `assert_fn`, the number of parameters should be the same
+        self.assertEqual(len(state_base), len(state_new))
+        assert_fn(state_base, state_new)
+
+    def _compare_models(
+        self, model, model_new, assert_fn, check_fp16=False, check_buffers=False
```

Comment:
qq: is there an issue with setting `check_buffers=True` by default to enable it for all tests that use this helper?

Reply:
Great idea! I was being a little overcautious in my attempt to minimize the changes I introduced. As long as we add a guard to the buffer comparison in ` _compare_models` to ensure there are buffers to compare (many of the current test models don't have any) we can set `check_buffers=True` to the default.
Note, I'm only checking persistent buffers given this is a `state_dict` inspection but there is at least one line of code that checks for non-persistent buffers that won't be covered (maybe best for a different PR if we want to tackle)